tinyagentos/cluster/router.py

```
@@ -16,21 +21,37 @@ def __init__(self, cluster: ClusterManager, http_client: httpx.AsyncClient):
             try:
                 url = f"{worker.url.rstrip('/')}{path}"
                 if method == "GET":
                     resp = await self.http_client.get(url, timeout=timeout)
                 else:
                     resp = await self.http_client.post(url, json=body, timeout=timeout)
                 resp.raise_for_status()
+                # Success — clear any failure record for this worker.
+                if ft is not None:
+                    ft.record_success(worker.name)
                 return resp.json(), worker.name
             except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)